File: backend/src/main/java/com/ll/nbe342team8/domain/member/member/dto/MemberDto.java
package com.ll.nbe342team8.domain.member.member.dto;


import com.ll.nbe342team8.domain.member.member.entity.Member;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.HashMap;
import java.util.Map;

@Getter
@NoArgsConstructor
public class MemberDto {
   private Long id;
   private String oAuthId;
   private String name;
   private String email;
   private Member.MemberType memberType;
   private String profileImageUrl;

   public MemberDto(Member entity) {
       this.id = entity.getId();
       this.oAuthId = entity.getOAuthId();
       this.name = entity.getName();
       this.email = entity.getEmail();
       this.memberType = entity.getMemberType();
       this.profileImageUrl = entity.getProfileImageUrl();
   }

   public Map<String, Object> getAttributes() {
       Map<String, Object> attributes = new HashMap<>();
       attributes.put("id", this.id);
       attributes.put("oAuthId", this.oAuthId);
       attributes.put("name", this.name);
       attributes.put("email", this.email);
       attributes.put("memberType", this.memberType);
       attributes.put("profileImageUrl", this.profileImageUrl);
       return attributes;
   }

   public enum MemberType {
       USER,
       ADMIN
   }
}

File: backend/src/main/java/com/ll/nbe342team8/domain/member/member/dto/MemberDto.kt
package com.ll.nbe342team8.domain.member.member.dto

import com.ll.nbe342team8.domain.member.member.entity.Member
import lombok.Getter
import lombok.NoArgsConstructor


@Getter
@NoArgsConstructor
class MemberDto(entity: Member) {
    private val id: Long = entity.id
    private val oAuthId: String = entity.oAuthId
    private val name: String? = entity.name
    private val email: String = entity.email
    private val memberType: Member.MemberType = entity.memberType
    private val profileImageUrl: String = entity.profileImageUrl

    val attributes: Map<String, Any?>
        get() {
            val attributes: MutableMap<String, Any?> = HashMap()
            attributes["id"] = id
            attributes["oAuthId"] = oAuthId
            attributes["name"] = name
            attributes["email"] = email
            attributes["memberType"] = memberType
            attributes["profileImageUrl"] = profileImageUrl
            return attributes
        }

    enum class MemberType {
        USER,
        ADMIN
    }
}